n2v/methods/rmks.py

```diff
@@ -6,7 +6,521 @@
 
 import numpy as np
 from opt_einsum import contract
-from scipy.optimize import minimize
+import psi4
+import time
 
 class MRKS():
-    pass
\ No newline at end of file
+    """
+    Wavefunction to KS_potential method based on
+    [1] [PRL 115, 083001 (2015)],
+    [2] [J. Chem. Phys. 146, 084103 (2017)].
+
+    The XC potential is calculated on the grid
+    instead of on the potential basis set.
+    Whereas, the guide potential is still used and
+    plays the role of v_hartree.
+    And because of this, the gird for vxc for output
+    has to be specified beforehand.
+    
+    For CIWavefunction as input, make sure to turn on
+    option opdm and tpdm:
+        psi4.set_options({
+            "opdm": True,
+            "tpdm": True,
+            'REFERENCE': "RHF"
+            })
+
+    Attributes:
+    -----------
+    Vpot: psi4.core.VBase
+        V_potential that contains the info of DFT spherical grid.
+    npoint_DFT: int
+        number of points for DFT spherical grid.
+    vxc_hole_WF: np.ndarray
+        vxc_hole_WF on spherical grid. This is stored
+        because the calculation of this takes most time.
+    """
+    vxc_hole_WF = None
+
+    def _vxc_hole_quadrature(self, grid_info=None, atol=1e-6):
```

Comment:
Just like in the main mRKS function, it would be nice to know in the documentation what changes in the code when there is a RHFwfn and CIwfn. 

Considering that this is the most computationally demanding part, we could think about writing this in Fortran in the near future. Connecting the compiled code to python shouldn't be a huge hassle.

Reply:
Good point. And easier to add parallelization.